transport/UdpEndpoint.h

```diff
@@ -3,19 +3,28 @@
 #include "memory/PacketPoolAllocator.h"
 #include "transport/BaseUdpEndpoint.h"
 
+namespace emulator
+{
+class FakeEndpointFactory;
+}
+
 namespace transport
 {
+class EndpointFactoryImpl;
+
 // end point that can be shared by multiple transports and can route incoming traffic
 class UdpEndpoint : public BaseUdpEndpoint
 {
-public:
+    friend class EndpointFactoryImpl;
+    friend class emulator::FakeEndpointFactory;
     UdpEndpoint(jobmanager::JobManager& jobManager,
```

Comment:
I don't see why the UdpEndpoint needs to have a private constructor and some `friends`. IMHO an object should not "baby-sitting" the other objects to prevent them from mistakes that are not of the responsibility of this object.

Also I didn't find the usage of `UdpEndpoint` on `FakeEndpointFactory`.

Reply:
The latter is correct, there is no use of `UdpEndpoint` from the `FakeEndpointFactory` -  it's remnant of the development, and I removed it.

Regarding making constructor private - that's the only way to enforce use of the factory, which, in-turn, is the only way to guarantee that all endpoints are created of the same type (either real, or fake).